tests/cpp/test_matmul_scheduler.cpp

```diff
@@ -2699,6 +2699,369 @@ TEST_F(MatmulSchedulerTest, DISABLED_RequireExternalPlugin) {
   MatmulParams params;
 }
 
+class AllocationDomainTest
+    : public NVFuserFixtureParamTest<std::tuple<bool, bool>> {
+ protected:
+  // Allocation order set by the pass breaks matmul tests
+  // see issue https://github.com/NVIDIA/Fuser/issues/1810
+  AllocationDomainTest() : optimization_guard_(false) {
+    DisableOptionsGuard::getCurOptions().set(DisableOption::MatmulExprEval);
+  }
+
+ private:
+  preseg_passes::OptimizationPassGuard<preseg_passes::AllocationDomainPass>
+      optimization_guard_;
+
+  DisableOptionsGuard option_guard_;
+};
+
+// This tests fusions where inputs to a Matmul will have the root domains
+// [M, K] and [K, N], and all possible combinations of allocation domains.
+// Please note that inpout in B is transposed prior to creating a Mma op.
+TEST_P(AllocationDomainTest, BasicMatmul) {
+  NVFUSER_TEST_CUDA_ARCH_RANGE_GUARD(7, 5, 9, 0);
+  bool a_m_inner = std::get<0>(GetParam());
+  bool b_k_inner = std::get<0>(GetParam());
+
+  auto setupFusion = [](Fusion* fusion, TensorView* tv0, TensorView* tv1) {
+    fusion->addInput(tv0);
+    fusion->addInput(tv1);
+
+    // This has rfactor: {N, K}
+    auto tv1t = transpose(tv1);
+
+    // [M, N, K]
+    auto tv0b = broadcast(tv0, {false, true, false});
+    auto tv1b = broadcast(tv1t, {true, false, false});
+
+    auto tv2 = fusedMultiplySum(tv0b, tv1b, {2});
+    fusion->addOutput(tv2);
+  };
+
+  auto schedule = [](Fusion* fusion) {
```

Comment:
Is there a particular reason to use lambdas for the definition and scheduling here?